rpcs/ledgerService.go

```diff
@@ -229,24 +230,65 @@ func (ls *LedgerService) ListenForCatchupReq(reqs <-chan network.IncomingMessage
 // a blocking function for handling a catchup request
 func (ls *LedgerService) handleCatchupReq(ctx context.Context, reqMsg network.IncomingMessage) {
 	var res WsGetBlockOut
+	target := reqMsg.Sender.(network.UnicastPeer)
+	var respTopics network.Topics
+
+	if target.Version() == "1" {
+
+		defer func() {
+			ls.sendCatchupRes(ctx, target, reqMsg.Tag, res)
+		}()
+		var req WsGetBlockRequest
+		err := protocol.DecodeReflect(reqMsg.Data, &req)
+		if err != nil {
+			res.Error = err.Error()
+			return
+		}
+		res.Round = req.Round
+		encodedBlob, err := RawBlockBytes(ls.ledger, basics.Round(req.Round))
+
+		if err != nil {
+			res.Error = err.Error()
+			return
+		}
+		res.BlockBytes = encodedBlob
+		return
+	}
+	// Else, if version == 2.1
 	defer func() {
-		ls.sendCatchupRes(ctx, reqMsg.Sender.(network.UnicastPeer), reqMsg.Tag, res)
+		target.Respond(ctx, reqMsg, respTopics)
 	}()
 
-	var req WsGetBlockRequest
-	err := protocol.DecodeReflect(reqMsg.Data, &req)
+	topics, err := network.UnmarshallTopics(reqMsg.Data)
 	if err != nil {
-		res.Error = err.Error()
+		errMsg := "LedgerService handleCatchupReq UnmarshallTopics error: " + err.Error()
+		topics = network.Topics{
+			network.MakeTopic(network.ErrorKey, []byte(errMsg))}
 		return
 	}
-	res.Round = req.Round
-	encodedBlob, err := RawBlockBytes(ls.ledger, basics.Round(req.Round))
-
-	if err != nil {
-		res.Error = err.Error()
+	roundBytes, found := topics.GetValue(roundKey)
+	if !found {
+		topics = network.Topics{
+			network.MakeTopic(network.ErrorKey,
+				[]byte("LedgerService handleCatchupReq: round-number topic is missing"))}
 		return
 	}
-	res.BlockBytes = encodedBlob
+	requestType, found := topics.GetValue(requestDataTypeKey)
+	if !found {
+		topics = network.Topics{
```

Comment:
changing the `topic` here won't help much. I think that you meant to change the `respTopics`.
( same goes for the rest of the errors )